lib/dialects/abstract/query-generator.js

```diff
@@ -933,12 +960,18 @@ const QueryGenerator = {
             )
           + ')', mainTable.as));
       } else {
-        mainQueryItems.push(this.selectFromTableFragment(options, mainTable.model, attributes.main, mainTable.quotedName, mainTable.as));
+        mainQueryItems.push(this.selectFromTableFragment(options, model, attributes.main, mainTable.quotedName, mainTable.as));
+      }
+
+      if (cteSelect) {
+        mainQueryItems.push(` INNER JOIN ${cteAs} ON `);
```

Comment:
I am going to rework `cteSelect` a bit. While an `INNER JOIN` does work, you are right that it does not make the most sense in this case.